backend/de.metas.business/src/main/java/de/metas/project/budget/BudgetProjectResources.java

```diff
@@ -100,4 +102,12 @@ public BudgetProjectResource getBudgetById(@NonNull final BudgetProjectResourceI
 				.orElseThrow(() -> new AdempiereException("No budget found for " + id));
 	}
 
+	@NonNull
+	public Optional<BudgetProjectResource> findBudget(@NonNull final ResourceId resourceId)
+	{
+		return budgets.stream()
+				.filter(budgetProjectResource -> budgetProjectResource.getBudgetProjectResourceData().getResourceId() != null)
```

Comment:
remark, as I described above, now all the code as poluted by `budgetProjectResource.getBudgetProjectResourceData()`.